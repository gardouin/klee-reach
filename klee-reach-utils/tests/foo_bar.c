#include "klee/klee.h"

#define VALUE 144

int foo(int x) {
	int i = VALUE;
	while (i > x) {
		i--;
	}
	return i;
}

int bar(int x, int y) {
	int a = foo(x);
	int b = foo(y);
	if (a == b) {
		klee_reach();
		return 1; // target here
	}
	return 0;
}

void foobar() {
	bar(0,0);
}

int main() {
	int x;
	int y;
	klee_make_symbolic(&x, sizeof(x), "x");
	klee_make_symbolic(&y, sizeof(y), "y");
	bar(x, y);
	return 0;
}
