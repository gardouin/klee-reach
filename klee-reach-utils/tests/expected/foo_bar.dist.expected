63:0
60:1
21:10
20:11
17:12
29:12
16:13
28:13
15:14
27:14
14:15
26:15
13:16
12:17
11:18
55:19
59:2
54:20
53:21
52:22
58:3
51:36
50:37
49:38
48:39
57:4
47:40
46:41
45:42
44:43
43:44
42:45
41:46
75:47
93:47
92:48
91:49
56:5
90:50
89:51
88:52
87:53
86:54
85:55
84:56
83:57
82:58
81:59
33:6
32:7
23:8
22:9
